src/components/Dialogs/StakingDialog/EnterStakingAmount.tsx

```diff
@@ -62,6 +72,39 @@ export const EnterStakingAmount = ({
             />
           ))}
         </div>
+        {/* Custom pool entry */}
+        <div className="mb-6">
+          <div className="text-sm text-[#9D9FA1] mb-2">
+            Or enter a custom staking pool
+          </div>
+          <div className="flex gap-2 items-center">
+            <Input
+              type="text"
+              placeholder="staking-pool.account.near"
+              value={customPoolId}
+              onChange={(e) => setCustomPoolId(e.target.value.trim())}
```

Comment:
This is going to trigger a LOT of network calls to check if the pool is whitelisted. My recommendation would be to either:
1. Add debouncing so we don't make a network request on _every_ keystroke OR
2. Only check if it's whitelisted when submitting the staking pool (and show an error if it's not whitelisted)